username.rb

```diff
@@ -2,13 +2,93 @@
 # Run `rspec /spec/username_spec.rb` to get started.
 
 def format_name(first, last)
-  nil
+	user_name = ""
+	if !first.empty? && !last.empty?
+	  	user_name = first.strip[0].downcase + last.strip.split.join('')
+	else
+		return nil
+	end
+	user_name.downcase
 end
 
-def format_year
-  nil
+def format_year(num)
+	if (num.to_s.length-1) == 3
+		return num.to_s[2,3]
+	else
+		return nil
+	end
 end
 
-def build_username
-  nil
+def check_privilege(num = 0)
+	num = num.floor
+	if num == 0
+		return 'user'
+	elsif num == 1
+		return 'seller'
+	elsif num == 2
+		return 'manager'
+	elsif num >= 3
+		return 'admin'
+	end
+end
+
+def user_type_prefix (num)
+	if num == 0
```

Comment:
I'm getting some code smell - you've already written this code before. How can you refactor this code by calling the check_privilege method?